src/main/java/com/zendesk/maxwell/MaxwellUpdateRowsEvent.java

```diff
@@ -48,4 +52,71 @@ public String sqlOperationString() {
 	public String getType() {
 		return "update";
 	}
+
+	private LinkedList<Pair<Row>> filteredRowsBeforeAndAfter;
+	private boolean performedBeforeAndAfterFilter;
+
+	private List<Pair<Row>> filteredRowsBeforeAndAfter() {
+		if ( this.filter == null)
+			return event.getRows();
+
+		if ( performedBeforeAndAfterFilter )
+			return filteredRowsBeforeAndAfter;
+
+		filteredRowsBeforeAndAfter = new LinkedList<>();
+		for ( Pair<Row> p : event.getRows()) {
+			if ( this.filter.matchesRow(this, p.getAfter()) )
+				filteredRowsBeforeAndAfter.add(p);
+		}
+		performedBeforeAndAfterFilter = true;
+		return filteredRowsBeforeAndAfter;
+	}
+
+	@Override
+	public List<RowMap> jsonMaps() {
+		ArrayList<RowMap> list = new ArrayList<>();
+		Object afterValue;
+		Object beforeValue;
+		for (Pair<Row> p : filteredRowsBeforeAndAfter() ) {
+			Row after = p.getAfter();
+			Row before = p.getBefore();
+
+			RowMap rowMap = buildRowMap();
+
+			Iterator<Column> aftIter = after.getColumns().iterator();
+			Iterator<Column> befIter = before.getColumns().iterator();
+			Iterator<ColumnDef> defIter = table.getColumnList().iterator();
+			while ( aftIter.hasNext() && defIter.hasNext() && befIter.hasNext() ) {
+				Column afterColumn = aftIter.next();
+				ColumnDef columnDef = defIter.next();
+				Column beforeColumn = befIter.next();
+
+				afterValue = valueForJson(afterColumn);
+				if ( afterValue != null ) {
+					afterValue = columnDef.asJSON(afterValue);
+				}
+
+				beforeValue = valueForJson(beforeColumn);
+				if ( beforeValue != null) {
+					beforeValue = columnDef.asJSON(beforeValue);
+				}
+
+				if ( afterValue != null && beforeValue != null ) {
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#equals(java.lang.Object,%20java.lang.Object)

I think you can replace this all with the null-safe Objects compare.
```java
if ( !Objects.equals(beforeValue, afterValue) ) {
  rowMap.putOldData(columnDef.getName(), beforeValue);
}
```